### PR TITLE
scheduler deprecates non-csi volumelimit plugins

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind deprecation

#### What this PR does / why we need it:

See https://github.com/kubernetes/kubernetes/pull/124003#discussion_r1574628509

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
kube-scheduler deprecated all non-csi volumelimit plugins and removed those from defaults plugins. 
- AzureDiskLimits
- CinderLimits
- EBSLimits
- GCEPDLimits

The NodeVolumeLimits plugin can handle the same functionality as the above plugins since the above volume types are migrated to CSI.
Please remove those plugins and replace them with the NodeVolumeLimits plugin if you explicitly use those plugins in the scheduler config.
Those plugins will be removed in the release 1.32.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by carlory** [comment]
cc @jsafrane

File: pkg/scheduler/apis/config/validation/validation.go [hunk 1]
```diff
@@ -101,7 +101,12 @@ type invalidPlugins struct {
 var invalidPluginsByVersion = []invalidPlugins{
 	{
 		schemeGroupVersion: v1.SchemeGroupVersion.String(),
-		plugins:            []string{},
```
Comment (kerthcet) [inline pkg/scheduler/apis/config/validation/validation.go:104]: Plz add a test to validate this, thanks. Otherwise LGTM

**Comment by kerthcet** [comment]
Also mention that we'll totally remove these plugins in release 1.32.

File: pkg/scheduler/apis/config/validation/validation.go (hunk 1 above)
Comment (carlory) [reply pkg/scheduler/apis/config/validation/validation.go:104]: @kerthcet  added.

**Review by sanposhiho — COMMENTED** [review]
/lgtm

**Comment by sanposhiho** [comment]
@carlory 

> Please just remove those plugins if you explicitly use those plugins in the scheduler config.

nit on release note: it's better to rephrase to suggest people **replacing** those plugins with NodeVolumeLimits plugin.

(People using plugins explicitly likely have to explicitly enable NodeVolumeLimits plugin.)

**Comment by carlory** [comment]
@sanposhiho the NodeVolumeLimits plugin is enabled by default. https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/apis/config/v1/default_plugins.go#L45

**Comment by sanposhiho** [comment]
Yes, I know. I meant, people enabling plugins explicitly are likely using: 

```yaml
enabled:
  - name: AzureDiskLimits
  - .....
disabled:
  - name: '*'
```

**Comment by jsafrane** [comment]
/lgtm
from storage point of view

**Comment by carlory** [comment]
> Yes, I know. I meant, people enabling plugins explicitly are likely using:
> 
> ```yaml
> enabled:
>   - name: AzureDiskLimits
>   - .....
> disabled:
>   - name: '*'
> ```
> 
> then, they have to explicitly enable NodeVolumeLimits

@sanposhiho release note is updated

```
Please remove those plugins and replace them with the NodeVolumeLimits plugin
```

**Review by sanposhiho — APPROVED** [review]
/unhold
/lgtm
/approve

**Pull request merged**